Week2/todo_app/program.js

```diff
@@ -62,11 +126,68 @@ dbConnection.connect(function(err) {
     console.log('connected as id ' + dbConnection.threadId);
 
     const todoModel = new TodoModel(dbConnection);
-    todoModel.load(function(err, todoItems) {
-        if(err) {
+    //Load all TODOs
+    todoModel.load(function (err, todoItems) {
+        if (err) {
             console.log("error loading TODO items:", err);
         }
+        console.log("existing todo items:", todoItems);
+    });
+
+    //create new TODO
+    todoModel.create(1, "Gaorieh Begin with week2", function (err, todoItems) {
+        if (err) {
+            console.log("error loading TODO items:", err);
+        }
+        console.log("existing todo items:", todoItems);
+    });
+
+    //update Todo by ID
+    todoModel.update(50, "Gaorieh NEW Todo ITEM", function (err, todoItems) {
+        if (err) {
+            console.log("error loading TODO items:", err);
+        }
+        console.log("existing todo items:", todoItems);
+    });
 
+    //delete existing Todo by ID
+    todoModel.delete(45, function (err, todoItems) {
+        if (err) {
+            console.log("error loading TODO items:", err);
+        }
         console.log("existing todo items:", todoItems);
     });
+
+    //tag Todo Item
+    todoModel.tagTodoItem(50, 2, function (err, todoItems) {
+        if (err) {
+            console.log("error loading TODO items:", err);
+        }
+        console.log("existing todo items:", todoItems);
+    });
+
+    //unTag Todo Item
+    todoModel.untagTodoItem(50, 2, function (err, todoItems) {
+        if (err) {
+            console.log("error loading TODO items:", err);
+        }
+        console.log("existing todo items:", todoItems);
+    });
+
+    //mark TODO as completed
+    todoModel.markCompleted(44, function (err, todoItems) {
+        if (err) {
+            console.log("error loading TODO items:", err);
+        }
+        console.log("existing todo items:", todoItems);
+    });
+
+    //mark TODO as Uncompleted
+    todoModel.markUnCompleted(44, function (err, todoItems) {
+        if (err) {
+            console.log("error loading TODO items:", err);
+        }
+        console.log("existing todo items:", todoItems);
+    });
+
```

Comment:
You didn't close your database connection.

Reply:
i added 
dbconnection.end();